client/goal.c

```diff
@@ -324,23 +324,18 @@ TDNFGoal(
     }
 
     queue_init(&queueJobs);
-    if(nAlterType == ALTER_UPGRADEALL)
-    {
+    if (nAlterType == ALTER_UPGRADEALL) {
         dwError = SolvAddUpgradeAllJob(&queueJobs);
-        BAIL_ON_TDNF_ERROR(dwError);
-    }
-    else if(nAlterType == ALTER_DISTRO_SYNC)
-    {
+    } else if(nAlterType == ALTER_DISTRO_SYNC) {
         dwError = SolvAddDistUpgradeJob(&queueJobs);
-        BAIL_ON_TDNF_ERROR(dwError);
+    } else if (pQueuePkgList->count == 0) {
```

Comment:
this does change the logic a bit from what it was. i would like to keep as is. 
what i mean is to do
else
{
    if (pQueuePkgList->count == 0)
    {
        dwError = ERROR_TDNF_ALREADY_INSTALLED;
        BAIL_ON_TDNF_ERROR(dwError);
    }
    for...
}

Reply:
AFAIK, this doesn't change the logic flow. Can you please explain what is the logical error here? I'm not able to find it out

Reply:
Lets say nAlterType was ALTER_UPGRADEALL for eg. In the previous version of the code, it would have never attempted the pQueuePkgList iteration and add to goal. Now it will regardless of what nAlterType is. That is a logic change.
From the context of this function, You do not control pQueuePkgList contents as it is an argument to this function. Therefore, the future behavior is also altered (inadvertently) if pQueuePkgList has contents.

Reply:
Sigh, that's so obvious, sorry :(

Reply:
Nothing to be sorry about. Appreciate your help